framework/src/main/java/org/checkerframework/common/returnsreceiver/qual/This.java

```diff
@@ -0,0 +1,39 @@
+package org.checkerframework.common.returnsreceiver.qual;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.checkerframework.framework.qual.PolymorphicQualifier;
+
+/**
+ * Write {@code @This} on the return type of a method that always returns its receiver ({@code
+ * this}). For example:
+ *
+ * <pre><code>
+ * class MyBuilder {
+ *   &#064;This MyBuilder setName(String name) {
+ *     this.name = name;
+ *     return this;
+ *   }
+ * }
+ * </code></pre>
+ *
+ * Strictly speaking, this is a polymorphic annotation, but when you write it on a return type, the
+ * Returns Receiver Checker automatically adds it to the receiver, so the above method is equivalent
+ * to:
+ *
+ * <pre><code>
+ * &#064;This MyBuilder setName(@This MyBuilder this, String name) {
+ *   this.name = name;
+ *   return this;
+ * }
+ * </code></pre>
+ *
+ * @checker_framework.manual #returns-receiver-checker Returns Receiver Checker
+ * @checker_framework.manual #qualifier-polymorphism Qualifier polymorphism
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE_USE, ElementType.TYPE_PARAMETER})
```

Comment:
You could meta-annotate all your qualifiers as `org.checkerframework.framework.qual.TargetLocations` for `RETURN` and `RECEIVER`. The annotation isn't enforced yet, but it might already be good documentation.

Reply:
It is also allowed on a downcast.  Would that be prohibited if we add the `@TargetLocations` meta-annotation?

Reply:
I went ahead and added the `@TargetLocations` annotation, under the assumption that a qualifier is always allowed to appear in a downcast